### PR TITLE
refactor broadcasting to support `except` and `restrict`

### PR DESCRIPTION
**Pull request by hhaensel**
Currently, it is not possible to address single clients by push! or run commands.

To remedy this situation I propose to
- modify existing keyword `except` to support not only UInt, but also Vetor{UInt} to exclude more than one client
- remove `WebSocket` as a possible value for except and rather supply the id as it is done in WebThreads
- add a keyword `restrict` to restrict messages to a client or list of clients

Furthermore, I have removed duplicate code in WebThreads.

There's a [corresponding PR in Stipple](https://github.com/GenieFramework/Stipple.jl/pull/244).

### CODE REVIEW

File: src/WebThreads.jl [hunk 1]
```diff
@@ -216,41 +216,28 @@ end
 """
 Pushes `msg` (and `payload`) to all the clients subscribed to the channels in `channels`.
 """
-function broadcast(channels::Union{ChannelName,Vector{ChannelName}}, msg::String;
-                    except::Union{HTTP.WebSockets.WebSocket,Nothing,UInt} = nothing) :: Bool
-  isa(channels, Array) || (channels = ChannelName[channels])
+function broadcast(channels::Union{ChannelName,Vector{ChannelName}}, msg::String, payload::Union{Dict,Nothing} = nothing;
```
Comment (essenciary) [inline src/WebThreads.jl:219]: Why is `payload` an argument now? It doesn't seem to be used.

Comment (essenciary) [reply src/WebThreads.jl:219]: Never mind, the diff is a bit hard to follow.

**Review by essenciary — APPROVED** [review]

File: src/WebThreads.jl (hunk 1 above)
Comment (hhaensel) [reply src/WebThreads.jl:219]: There were two methods with almost identical code except that payload was used in one of them.
I merged them in one method just in the same way you had done this for WebChannels.
